shared/desktop/webpack.config.babel.js

```diff
@@ -0,0 +1,320 @@
+        test: /\.jsx?$/,
+        use: [babelRule],
+      },
+      {
+        test: [/emoji-datasource.*\.(gif|png)$/, /\.ttf$/],
+        use: [fileLoaderRule],
+      },
+      {
+        test: /\.css$/,
+        use: ['style-loader', 'css-loader'],
+      },
+    ]
+  }
+
+  const makeCommonPlugins = () => {
+    const defines = {
+      __DEV__: flags.isDev,
+      __HOT__: JSON.stringify(flags.isHot),
+      __SCREENSHOT__: flags.isVisDiff,
+      __VERSION__: flags.isDev ? JSON.stringify('Development') : JSON.stringify(process.env.APP_VERSION),
+      'process.env.NODE_ENV': flags.isDev ? JSON.stringify('development') : JSON.stringify('production'),
+    }
+    console.warn('Injecting defines: ', defines)
+    const definePlugin = [new webpack.DefinePlugin(defines)]
+
+    const uglifyPlugin = flags.isDev
+      ? []
+      : [
+          new UglifyJSPlugin({
+            comments: false,
+            sourceMaps: true,
+          }),
+        ]
+
+    return [...definePlugin, ...uglifyPlugin].filter(Boolean)
+  }
+
+  // If we use the hot server it pulls in this config
+  const devServer = {
+    compress: true,
+    contentBase: path.resolve(__dirname, 'dist'),
+    hot: flags.isHot,
+    lazy: false,
+    overlay: true,
+    port: 4000,
+    publicPath: 'http://localhost:4000/dist/',
+    quiet: false,
+    stats: {
+      colors: true,
+    },
+  }
+
+  return {
+    bail: true,
+    cache: flags.isDev,
+    devServer,
+    module: {
+      rules: makeRules(),
+    },
+    node: {
+      __dirname: true,
+    },
+    output: {
+      filename: '[name].bundle.js',
+      path: path.resolve(__dirname, 'dist'),
+      publicPath: flags.isHot ? 'http://localhost:4000/dist/' : '../dist/',
+    },
+    plugins: makeCommonPlugins(),
+    resolve: {
+      extensions: ['.desktop.js', '.js', '.jsx', '.json', '.flow'],
+    },
+  }
+}
+
+const makeMainThreadConfig = () => {
+  const makeEntries = () => {
+    if (flags.isVisDiff) {
+      return {
+        'render-visdiff': path.resolve(__dirname, 'test/render-visdiff.js'),
+      }
+    } else {
+      return {
+        main: path.resolve(__dirname, 'app/index.js'),
+      }
+    }
+  }
+
+  return merge(commonConfig, {
+    entry: makeEntries(),
+    name: 'mainThread',
+    target: 'electron-main',
+  })
+}
+
+const makeRenderThreadConfig = () => {
+  const makeRenderPlugins = () => {
+    // Visual dashboard to see what the hot server is doing
+    const dashboardPlugin = flags.isShowingDashboard ? [new DashboardPlugin()] : []
+    // Allow hot module reload when editing files
+    const hmrPlugin = flags.isHot && flags.isDev
+      ? [new webpack.HotModuleReplacementPlugin(), new webpack.NamedModulesPlugin()]
+      : []
+    // Don't spit out errors while building
+    const noEmitOnErrorsPlugin = flags.isDev ? [new webpack.NoEmitOnErrorsPlugin()] : []
+    // Put common code between the entries into a sep. file
+    const commonChunksPlugin = flags.isDev
```

Comment:
this is new, we pull common code from the various entypoints (menubar, tracker popups etc) into common-chunks.js